src/components/v5/common/Fields/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,171 @@
+import clsx from 'clsx';
+import isAfter from 'date-fns/isAfter';
+import React, { useCallback, useRef, useState } from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+import { useMobile } from '~hooks';
+import { formatText } from '~utils/intl.ts';
+import Button from '~v5/shared/Button/index.ts';
+import { type ButtonMode } from '~v5/shared/Button/types.ts';
+
+import { DATEPICKER_PORTAL_ID, DEFAULT_DATE_FORMAT } from './consts.ts';
+import DatepickerContainer from './partials/DatepickerContainer/index.ts';
+import DatepickerCustomHeader from './partials/DatepickerCustomHeader/index.ts';
+import DatepickerCustomInput from './partials/DatepickerCustomInput/index.ts';
+import DatepickerCustomRangeInput from './partials/DatepickerCustomRangeInput/index.ts';
+import { type DatepickerProps } from './types.ts';
+
+import styles from './Datepicker.css';
+
+const Datepicker = <T extends boolean = false>({
```

Comment:
In `react-datepicker` they use `WithRange` as the name of this generic, could we rename it here too? It'd be easier to figure out what it does